api/apps/geoprocessing/src/modules/cleanup-tasks/cleanup-tasks.service.ts

```diff
@@ -0,0 +1,232 @@
+            });
+          return {
+            projectId: p.project_id,
+            data: { projectCustomFeaturesIds },
+          };
+        }),
+      );
+
+      // Uses unused resources cleanup service to finish cleaning for every mismatch in projectId array
+      await notMatchingProjectsIdWithCustomFeaturesPerProject.map(async (p) => {
+        await this.projectUnusedResources.removeUnusedResources(
+          p.projectId,
+          p.data,
+        );
+        return p;
+      });
+
+      await this.geoEntityManager.query(
+        `TRUNCATE TABLE project_nuke_preparation`,
+      );
+    });
+  }
+
+  async nukeScenariosDanglingData() {
+    const scenarioIds = await this.apiEntityManager
+      .createQueryBuilder()
+      .from('scenarios', 'p')
+      .select(['id'])
+      .getRawMany()
+      .then((result) => result.map((i) => i.id))
+      .catch((error) => {
+        throw new Error(error);
+      });
+
+    await this.geoEntityManager.transaction(async (entityManager) => {
+      await this.geoEntityManager.query(
+        `TRUNCATE TABLE scenario_nuke_preparation`,
+      );
+
+      for (const [, summaryChunks] of chunk(
+        scenarioIds,
+        CHUNK_SIZE_FOR_BATCH_DB_OPERATIONS,
+      ).entries()) {
+        await entityManager.insert(
+          'scenario_nuke_preparation',
+          summaryChunks.map((chunk: string) => ({
+            scenario_id: chunk,
+          })),
+        );
+      }
+
+      const missingScenarioIdsFromScenarioFeaturesData: entitiesWithScenarioId[] = await this.geoEntityManager.query(
+        `SELECT sfd.scenario_id
+          FROM scenario_features_data sfd
+          LEFT JOIN scenario_nuke_preparation snp ON snp.scenario_id = sfd.scenario_id
+          WHERE snp.scenario_id IS NULL`,
+      );
+      const missingScenarioIdsFromBlmFinalResults: entitiesWithScenarioId[] = await this.geoEntityManager.query(
+        `SELECT bfr.scenario_id
+          FROM blm_final_results bfr
+          LEFT JOIN scenario_nuke_preparation snp ON snp.scenario_id = bfr.scenario_id
+          WHERE snp.scenario_id IS NULL`,
+      );
+      const missingScenarioIdsFromBlmPartialResults: entitiesWithScenarioId[] = await this.geoEntityManager.query(
+        `SELECT bpr.scenario_id
+          FROM blm_final_results bpr
+          LEFT JOIN scenario_nuke_preparation snp ON snp.scenario_id = bpr.scenario_id
+          WHERE snp.scenario_id IS NULL`,
+      );
+      const missingScenarioIdsFromMarxanExecutionMetadata: entitiesWithScenarioId[] = await this.geoEntityManager.query(
+        `SELECT mem.scenarioId
+          FROM marxan_execution_metadata mem
+          LEFT JOIN scenario_nuke_preparation snp ON snp.scenario_id = mem.scenarioId
+          WHERE snp.scenario_id IS NULL`,
+      );
+      const missingScenarioIdsFromScenariosPuData: entitiesWithScenarioId[] = await this.geoEntityManager.query(
+        `SELECT spd.scenario_id
+          FROM scenarios_pu_data spd
```

Comment:
But all this cleanup is done just in case something gets wonky and on cascade do not delete anything properly, right? I mean, other tables I referenced here have also on cascades and I am cleaning them up anyways, right?

Reply:
in general, we should trust postgresql on cascades - if there are other entities that will be deleted via cascades and we're explicitly deleting them here, then we should avoid doing this (it may create more trouble than it tries to solve, especially if done outside of a db transaction) - I checked the entities we are deleting in your original implementation last week and IIRC this was the only one that was already covered by cascades.

Reply:
tl;dr basically this task is about manually triggering deletes that would be left to sql cascades _if all the data were to be in a single db_, but when we can rely on db cascades within `geodb`, then we should let postgresql work for us 😃